### PR TITLE
[FLINK-12402][table] Make validation error message for CallExpression more user friendly

### PR DESCRIPTION
**Pull request by hequn8128**
## What is the purpose of the change

This pull request makes the validation error message more user friendly. The previous error message for CallExpression validation may not display the root cause which may confuse our users. See more details in the [jira](https://issues.apache.org/jira/browse/FLINK-12402).

## Brief change log

  - Display the truth validation error message.
  - Adjust test cases.

## Verifying this change

This change added tests and can be verified as follows:

  - Add `testInvalidParameterTypes` in `CalcValidationTest` to test the validation error message.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

**Comment by dawidwys** [comment]
@flinkbot approve-until architecture

File: flink-table/flink-table-planner/src/main/java/org/apache/flink/table/expressions/rules/ResolveCallByArgumentsRule.java [hunk 1]
```diff
@@ -86,21 +86,31 @@ private Expression performTypeResolution(CallExpression call, List<TypeInformati
...
-
 			return call;
 		}
 
+		/**
+		 * Return the validation error message of this {@link PlannerExpression} or return the
+		 * validation error message of it's children if it passes the validation. Return an empty
+		 * string if all validation succeeded.
+		 */
+		private String getValidationErrorMessage(PlannerExpression plannerCall) {
```
Comment (dawidwys) [inline flink-table/flink-table-planner/src/main/java/org/apache/flink/table/expressions/rules/ResolveCallByArgumentsRule.java]: One small comment, let's use the `Optional` to mark that method might not return results.

**Comment by hequn8128** [comment]
Hi @dawidwys , thanks a lot for your review and suggestions. I have addressed your comments and updated the PR.

Best, Hequn

**Comment by hequn8128** [comment]
@dawidwys thank you for your review and merge.

**Pull request merged**